docs/high-level-overview.md

```diff
@@ -1,165 +1,179 @@
-allow transaction de-duplication to be done at the transaction pool level and also remove the cost
-for message relayers to run header relayers.
+Offchain bridge components are separate processes, called relayers. Relayers are connected both to the
+source chain and target chain nodes. Relayers are reading state of the source chain, compare it to the
+state of the target chain and, if state at target chain needs to be updated, submits target chain
+transaction.
 
-### Message Passing
+### GRANDPA Finality Relay
 
-Once header sync is maintained, the target side of the bridge can receive and verify proofs about
-events happening on the source chain, or its internal state. On top of this, we built a message
-passing protocol which consists of two parts described in following sections: message delivery and
-message dispatch.
+The task of relay is to submit source chain GRANDPA justifications and their corresponding headers to
+the Bridge GRANDPA Finality Pallet, deployed at the target chain. For that, the relay subscribes to
+the GRANDPA justifications stream and submits every new justification it sees. In addition, relay is
+searching for mandatory headers and submits their justifications - without that the pallet will be
+unable to move forward.
 
-#### Message Lanes Delivery
+More: [GRANDPA Finality Relay Sequence Diagram](./grandpa-finality-relay.html), [code](../relays/finality/).
 
-The [Message delivery pallet](../modules/messages/src/lib.rs) is responsible for queueing up
-messages and delivering them in order on the target chain. It also dispatches messages, but we will
-cover that in the next section.
+### Parachains Finality Relay
 
-The pallet supports multiple lanes (channels) where messages can be added. Every lane can be
-considered completely independent from others, which allows them to make progress in parallel.
-Different lanes can be configured to validated messages differently (e.g higher rewards, specific
-types of payload, etc.) and may be associated with a particular "user application" built on top of
-the bridge. Note that messages in the same lane MUST be delivered _in the same order_ they were
-queued up.
+The relay connects to the source _relay_ chain and the target chain nodes. It doesn't need to connect to the
+tracked parachain nodes. The relay looks at the [`Heads`](https://github.com/paritytech/polkadot/blob/1a034bd6de0e76721d19aed02a538bcef0787260/runtime/parachains/src/paras/mod.rs#L642)
+map of the [`paras` pallet](https://github.com/paritytech/polkadot/tree/1a034bd6de0e76721d19aed02a538bcef0787260/runtime/parachains/src/paras) and compares the value with the best parachain head, stored at the bridge parachains pallet at
+the taget chain. If new parachain head appears at the relay chain block `B`, the relay process **waits**
+until header `B` or one of its ancestors appears at the target chain. Once it is available, the storage
+proof of the map entry is generated and is submitted to the target chain.
```

Comment:
I think it waits for `B` or _children_ of `B` (block numbers > B), rather than _ancestors_ (parents) of `B`... right?

```suggestion
the taget chain.
When new parachain head appears at source relay chain block `B`, the relay process **waits**
until header `B` or one of its children blocks' finality is relayed to and appears on target chain.
Once it is available, the storage proof of the map entry is generated and is submitted to the
target chain.
```